libraries/AP_InertialSensor/AP_InertialSensor_SCHA63T.cpp

```diff
@@ -0,0 +1,451 @@
+
+AP_InertialSensor_SCHA63T::AP_InertialSensor_SCHA63T(AP_InertialSensor &imu,
+        AP_HAL::OwnPtr<AP_HAL::Device> _dev_uno,
+        AP_HAL::OwnPtr<AP_HAL::Device> _dev_due,
+        enum Rotation _rotation)
+    : AP_InertialSensor_Backend(imu)
+    , dev_uno(std::move(_dev_uno))
+    , dev_due(std::move(_dev_due))
+    , rotation(_rotation)
+{
+}
+
+AP_InertialSensor_Backend *
+AP_InertialSensor_SCHA63T::probe(AP_InertialSensor &imu,
+                                 AP_HAL::OwnPtr<AP_HAL::SPIDevice> dev_uno,
+                                 AP_HAL::OwnPtr<AP_HAL::SPIDevice> dev_due,
+                                 enum Rotation rotation)
+{
+    if (!dev_uno || !dev_due) {
+        return nullptr;
+    }
+    auto sensor = new AP_InertialSensor_SCHA63T(imu, std::move(dev_uno), std::move(dev_due), rotation);
+
+    if (!sensor) {
+        return nullptr;
+    }
+
+#if defined(HAL_GPIO_PIN_SCHA63T_RESET)
+    palSetLine(HAL_GPIO_PIN_SCHA63T_RESET);
+#endif
+
+    if (!sensor->init()) {
+        delete sensor;
+        return nullptr;
+    }
+
+    return sensor;
+}
+
+void AP_InertialSensor_SCHA63T::start()
+{
+    if (!_imu.register_accel(accel_instance, BACKEND_SAMPLE_RATE, dev_uno->get_bus_id_devtype(DEVTYPE_INS_SCHA63T)) ||
+        !_imu.register_gyro(gyro_instance, BACKEND_SAMPLE_RATE, dev_due->get_bus_id_devtype(DEVTYPE_INS_SCHA63T))) {
+        return;
+    }
+
+    // set backend rate
+    uint16_t backend_rate_hz = BACKEND_SAMPLE_RATE;
+    if (enable_fast_sampling(accel_instance) && get_fast_sampling_rate() > 1) {
+        bool fast_sampling = dev_uno->bus_type() == AP_HAL::Device::BUS_TYPE_SPI;
+        if (fast_sampling) {
+            // constrain the gyro rate to be a 2^N multiple
+            uint8_t fast_sampling_rate = constrain_int16(get_fast_sampling_rate(), 1, 4);
+            // calculate rate we will be giving samples to the backend
+            backend_rate_hz = constrain_int16(backend_rate_hz * fast_sampling_rate, backend_rate_hz, BACKEND_SAMPLE_RATE_MAX);
+        }
+    }
+    uint32_t backend_period_us = 1000000UL / backend_rate_hz;
+
+    // setup sensor rotations from probe()
+    set_gyro_orientation(gyro_instance, rotation);
+    set_accel_orientation(accel_instance, rotation);
+
+    // setup callbacks
+    dev_uno->register_periodic_callback(backend_period_us, FUNCTOR_BIND_MEMBER(&AP_InertialSensor_SCHA63T::read_accel, void));
+    dev_due->register_periodic_callback(backend_period_us, FUNCTOR_BIND_MEMBER(&AP_InertialSensor_SCHA63T::read_gyro, void));
+}
+
+/*
+  probe and initialise accelerometer
+ */
+bool AP_InertialSensor_SCHA63T::init()
+{
+    WITH_SEMAPHORE(dev_uno->get_semaphore());
+    WITH_SEMAPHORE(dev_due->get_semaphore());
+
+    // error initialise is OK
+    ret_scha63t = true;
+
+    // wait 25ms for non-volatile memory (NVM) read
+    hal.scheduler->delay(25);
+
+    // set DUE operation mode on (must be less than 1ms)
+    ret_scha63t &= RegisterWrite(SCHA63T_DUE, MODE, MODE_NORM);
```

Comment:
why call this twice?

Reply:
The sensor datasheet says to operate twice in the power up sequence.
![figure1](https://user-images.githubusercontent.com/113963003/231319975-2f194602-009b-4b69-b6bf-a18467e7e77b.png)